Enhancements for the HCP Terraform Operator for Kubernetes (#688)
hashicorp/terraform-docs-common: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx

> AgentDeploymentAutoscaling allows you to configure the operator
> to scale the deployment for an AgentPool up and down to meet demand.

Reviewer: ```suggestion
AgentDeploymentAutoscaling configures the operator to scale the deployment for an AgentPool up and down to meet demand.
```

| --- | --- |
| `maxReplicas` _integer_ | MaxReplicas is the maximum number of replicas for the Agent deployment. |
| `minReplicas` _integer_ | MinReplicas is the minimum number of replicas for the Agent deployment. |

> | `targetWorkspaces` _[TargetWorkspace](#targetworkspace)_ | TargetWorkspaces is a list of HCP Terraform Workspaces which<br />the agent pool should scale up to meet demand. When this field<br />is ommited the autoscaler will target all workspaces that are<br />associated with the AgentPool. |

Reviewer: why are we manually adding line breaks here?

Author: Cleaned up the line breaks

| `cooldownPeriodSeconds` _integer_ | CooldownPeriodSeconds is the time to wait between scaling events. Defaults to 300. |

> | `cooldownPeriod` _[AgentDeploymentAutoscalingCooldownPeriod](#agentdeploymentautoscalingcooldownperiod)_ | CoolDownPeriod configures the period to wait between scaling up and scaling down |

Reviewer: ```suggestion
| `cooldownPeriod` _[AgentDeploymentAutoscalingCooldownPeriod](#agentdeploymentautoscalingcooldownperiod)_ | CoolDownPeriod is the period to wait between scaling up and scaling down |
```

so it's consistent with the other attributes

#### AgentDeploymentAutoscalingCooldownPeriod

> 
> 
> 

Reviewer: ```suggestion

```

> AgentDeploymentAutoscalingCooldownPeriod configures the period to wait between scaling up and scaling down

Reviewer: ```suggestion
AgentDeploymentAutoscalingCooldownPeriod configures the period to wait between scaling up and scaling down.
```

| --- | --- |
| `apiVersion` _string_ | `app.terraform.io/v1alpha2`
| `kind` _string_ | `AgentPool`

> | `kind` _string_ | Kind is a string value representing the REST resource this object represents.<br />Servers may infer this from the endpoint the client submits requests to.<br />Cannot be updated.<br />In CamelCase.<br />More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds |

Reviewer: same thing here, why are we adding line breaks to this?

AgentPoolSpec defines the desired state of AgentPool.

_Appears in:_
- [AgentPool](#agentpool)

| Field | Description |
| --- | --- |
| `name` _string_ | Agent Pool name.<br />More information:<br />  - https://developer.hashicorp.com/terraform/cloud-docs/agents/agent-pools |

> | `organization` _string_ | Organization name where the Workspace will be created.<br />More information:<br />  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/organizations |

Reviewer: i think the previous one was ok?

#### AgentToken

> Agent Token is a secret token that a HCP Terraform Agent is used to connect to the HCP Terraform Agent Pool.
> In `spec` only the field `Name` is allowed, the rest are used in `status`.
> More infromation:
>   - https://developer.hashicorp.com/terraform/cloud-docs/agents

Reviewer: i think the previous wording was a lil bit more clear. i also updated the second sentence to be active voice and fixed typo

```suggestion
HCP Terraform agents use secret tokens, called `AgentToken`s, to connect to the HCP Terraform agent pool. In `spec`, you can only define `Name`. The other attributes are returned in `status`. 

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/agents
```

ConsumerWorkspace allows access to the state for specific workspaces within the same organization.
Only one of the fields `ID` or `Name` is allowed.
At least one of the fields `ID` or `Name` is mandatory.
More information:

>   - https://developer.hashicorp.com/terraform/cloud-docs/workspaces/state#remote-state-access-controls

Reviewer: is there a reason why we put these on three different lines? it renders the same way

<img width="728" alt="image" src="https://github.com/user-attachments/assets/347b31a4-8a98-4594-89d4-2ece3461e400">

Author: Updated this across the doc

_Appears in:_
- [RemoteStateSharing](#remotestatesharing)

| Field | Description |
| --- | --- |
| `id` _string_ | Consumer Workspace ID.<br />Must match pattern: `^ws-[a-zA-Z0-9]+$` |
| `name` _string_ | Consumer Workspace name. |

#### CustomPermissions

> Custom permissions let you assign specific, finer-grained permissions to a team than the broader fixed permission sets provide.
> More information:
>   - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/permissions#custom-workspace-permissions

Reviewer: ```suggestion
Custom permissions let you assign specific, finer-grained permissions to a team than the broader fixed permission sets provide.

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/permissions#custom-workspace-permissions
```

Reviewer: I'm not sure I like these full URLs, why are we switching to this?

Author: I'll do a quick pass to clean these link up

| `lockWorkspace` _boolean_ | Allows users to manually lock the workspace to temporarily prevent runs.<br />When a workspace's execution mode is set to "local", users must have this permission to perform local CLI runs using the workspace's state.<br />Default: `false`. |
| `runs` _[WorkspaceRunsPermissionType](#workspacerunspermissiontype)_ | Run access.<br />Must be one of the following values: `apply`, `plan`, `read`.<br />Default: `read`. |
| `runTasks` _boolean_ | Manage Workspace Run Tasks.<br />Default: `false`. |
| `sentinelMocks` _[WorkspaceSentinelMocksPermissionType](#workspacesentinelmockspermissiontype)_ | Download Sentinel mocks.<br />Must be one of the following values: `none`, `read`.<br />Default: `none`. |
| `stateVersions` _[WorkspaceStateVersionsPermissionType](#workspacestateversionspermissiontype)_ | State access.<br />Must be one of the following values: `none`, `read`, `read-outputs`, `write`.<br />Default: `none`. |
| `variables` _[WorkspaceVariablesPermissionType](#workspacevariablespermissiontype)_ | Variable access.<br />Must be one of the following values: `none`, `read`, `write`.<br />Default: `none`. |

#### Module

> Module is the Schema for the modules API
> Module implements the API-driven Run Workflow
> More information:
>   - https://developer.hashicorp.com/terraform/cloud-docs/run/api

Reviewer: ```suggestion
Module is the Schema for the modules API. Module implements the API-driven run workflow.

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/run/api
```

| Field | Description |
| --- | --- |
| `organization` _string_ | Organization name where the Workspace will be created.<br />More information:<br />  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/organizations |
| `token` _[Token](#token)_ | API Token to be used for API calls. |
| `module` _[ModuleSource](#modulesource)_ | Module source and version to execute. |
| `workspace` _[ModuleWorkspace](#moduleworkspace)_ | Workspace to execute the module. |
| `name` _string_ | Name of the module that will be uploaded and executed.<br />Default: `this`. |
| `variables` _[ModuleVariable](#modulevariable) array_ | Variables to pass to the module, they must exist in the Workspace. |
| `outputs` _[ModuleOutput](#moduleoutput) array_ | Module outputs to store in ConfigMap(non-sensitive) or Secret(sensitive). |
| `destroyOnDeletion` _boolean_ | Specify whether or not to execute a Destroy run when the object is deleted from the Kubernetes.<br />Default: `false`. |

> | `restartedAt` _string_ | Allows executing a new Run without changing any Workspace or Module attributes.<br />Example: kubectl patch KIND NAME --type=merge --patch '{"spec": {"restartedAt": "'\`date -u -Iseconds\`'"}}' |

Reviewer: ```suggestion
| `restartedAt` _string_ | Allows executing a new Run without changing any Workspace or Module attributes.<br />Example: `kubectl patch KIND NAME --type=merge --patch '{"spec": {"restartedAt": "'\`date -u -Iseconds\`'"}}'` |
```

currently:
<img width="710" alt="image" src="https://github.com/user-attachments/assets/e014f5eb-b488-435d-b0d3-5b54afe63b84">

| `runID` _string_ | Run ID of the latest run that updated the outputs. |

#### PlanStatus

> 
> 
> 
> 
> 

Reviewer: ```suggestion

```
